module 'ccxt.kraken' from ../site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'

Running `pytest` locally, I get:

```text
FAILED pdr_backend/trueval/test/test_trueval.py::test_get_trueval_success - AttributeError: <module 'ccxt.kraken' from '/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'
FAILED pdr_backend/trueval/test/test_trueval.py::test_get_trueval_fail - AttributeError: <module 'ccxt.kraken' from '/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'
FAILED pdr_backend/trueval/test/test_trueval_agent.py::test_submit_trueval_mocked_price_down - AttributeError: <module 'ccxt.kraken' from '/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'
FAILED pdr_backend/trueval/test/test_trueval_agent.py::test_submit_trueval_mocked_price_up - AttributeError: <module 'ccxt.kraken' from '/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'
```


### Full traceback

```text
_______________________________________________________________________________ test_submit_trueval_mocked_price_up _______________________________________________________________________________

agent = <pdr_backend.trueval.trueval_agent.TruevalAgent object at 0x119d16a90>, slot = <pdr_backend.models.slot.Slot object at 0x119d15e90>
predictoor_contract = <MagicMock name='PredictoorContract' id='4728124048'>

    def test_submit_trueval_mocked_price_up(agent, slot, predictoor_contract):
>       with patch("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_up):

pdr_backend/trueval/test/test_trueval_agent.py:41: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/opt/homebrew/Cellar/python@3.11/3.11.4_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/mock.py:1443: in __enter__
    original, local = self.get_original()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <unittest.mock._patch object at 0x119d14090>

    def get_original(self):
        target = self.getter()
        name = self.attribute
    
        original = DEFAULT
        local = False
    
        try:
            original = target.__dict__[name]
        except (AttributeError, KeyError):
            original = getattr(target, name, DEFAULT)
        else:
            local = True
    
        if name in _builtins and isinstance(target, ModuleType):
            self.create = True
    
        if not self.create and original is DEFAULT:
>           raise AttributeError(
                "%s does not have the attribute %r" % (target, name)
            )
E           AttributeError: <module 'ccxt.kraken' from '/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/ccxt/kraken.py'> does not have the attribute 'fetch_ohlcv'

/opt/homebrew/Cellar/python@3.11/3.11.4_1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/unittest/mock.py:1416: AttributeError
```

--- a/pdr_backend/publisher/test/test_publish.py
+++ b/pdr_backend/publisher/test/test_publish.py
@@ -22,10 +22,10 @@ def test_fund_dev_accounts(monkeypatch):
 
     fund_dev_accounts(accounts_to_fund, mock_account, mock_token)
 
-    address = Account.from_key(private_key=pk).address
+    a = Account.from_key(private_key=pk)  # pylint: disable=no-value-for-parameter
     mock_token.transfer.assert_has_calls(
         [
-            call(address, 2e21, mock_account),
-            call(address, 3e21, mock_account),
+            call(a.address, 2e21, mock_account),
+            call(a.address, 3e21, mock_account),
         ]
     )

--- a/pdr_backend/trueval/test/test_trueval.py
+++ b/pdr_backend/trueval/test/test_trueval.py
@@ -1,23 +1,11 @@
-from unittest.mock import patch
-from pdr_backend.trueval.main import get_trueval
+from enforce_typing import enforce_types
+
+from pdr_backend.trueval.trueval_agent import get_trueval
 from pdr_backend.models.feed import Feed
 
 
-def mock_fetch_ohlcv(*args, **kwargs):
-    since = kwargs.get("since")
-    if since == 1:
-        return [[None, 100]]
-    elif since == 2:
-        return [[None, 200]]
-    else:
-        raise ValueError("Invalid timestamp")
-
-
-def mock_fetch_ohlcv_fail(*args, **kwargs):
-    return []
-
-
-def test_get_trueval_success():
+@enforce_types
+def test_get_trueval_success(monkeypatch):
     feed = Feed(
         name="ETH-USDT",
         address="0x1",
@@ -31,11 +19,22 @@ def test_get_trueval_success():
         owner="0xowner",
     )
 
-    with patch("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv):
-        result = get_trueval(feed, 1, 2)
-        assert result == (True, False)  # 1st True because 200 > 100
+    def mock_fetch_ohlcv(*args, **kwargs):
+        since = kwargs.get("since")
+        if since == 1:
+            return [[None, 100]]
+        elif since == 2:
+            return [[None, 200]]
+        else:
+            raise ValueError("Invalid timestamp")
+
+    monkeypatch.setattr("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv)
+
+    result = get_trueval(feed, 1, 2)
+    assert result == (True, False)  # 1st True because 200 > 100
 
 
+@enforce_types
 def test_get_trueval_live_lowercase_slash():
     feed = Feed(
         name="ETH-USDT",
@@ -54,6 +53,7 @@ def test_get_trueval_live_lowercase_slash():
     assert result == (True, False)
 
 
+@enforce_types
 def test_get_trueval_live_lowercase_dash():
     feed = Feed(
         name="ETH-USDT",
@@ -72,7 +72,8 @@ def test_get_trueval_live_lowercase_dash():
     assert result == (True, False)
 
 
-def test_get_trueval_fail():
+@enforce_types
+def test_get_trueval_fail(monkeypatch):
     feed = Feed(
         name="ETH-USDT",
         address="0x1",
@@ -86,6 +87,10 @@ def test_get_trueval_fail():
         owner="0xowner",
     )
 
-    with patch("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_fail):
-        result = get_trueval(feed, 1, 2)
-        assert result == (False, True)  # 2nd True because failed
+    def mock_fetch_ohlcv_fail(*args, **kwargs):
+        return []
+
+    monkeypatch.setattr("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_fail)
+
+    result = get_trueval(feed, 1, 2)
+    assert result == (False, True)  # 2nd True because failed

--- a/pdr_backend/trueval/test/test_trueval_agent.py
+++ b/pdr_backend/trueval/test/test_trueval_agent.py
@@ -1,15 +1,20 @@
 from unittest.mock import patch, Mock, MagicMock
+
+from enforce_typing import enforce_types
 import pytest
+
 from pdr_backend.trueval.trueval_agent import TruevalAgent
 from pdr_backend.trueval.trueval_config import TruevalConfig
 from pdr_backend.trueval.trueval_agent import get_trueval
 
 
+@enforce_types
 def test_new_agent(trueval_config):
     agent = TruevalAgent(trueval_config, get_trueval)
     assert agent.config == trueval_config
 
 
+@enforce_types
 def test_process_slot(agent, slot, predictoor_contract):
     with patch.object(
         agent, "get_and_submit_trueval", return_value={"tx": "0x123"}
@@ -19,6 +24,7 @@ def test_process_slot(agent, slot, predictoor_contract):
         mock_submit.assert_called()
 
 
+@enforce_types
 def test_get_contract_info_caching(agent, predictoor_contract):
     agent.get_contract_info("0x1")
     agent.get_contract_info("0x1")
@@ -26,28 +32,31 @@ def test_get_contract_info_caching(agent, predictoor_contract):
     predictoor_contract.assert_called_once_with(agent.config.web3_config, "0x1")
 
 
-def test_submit_trueval_mocked_price_down(agent, slot, predictoor_contract):
-    with patch("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_down):
-        result = agent.get_and_submit_trueval(
-            slot, predictoor_contract.return_value, 60
-        )
-        assert result == {"tx": "0x123"}
-        predictoor_contract.return_value.submit_trueval.assert_called_once_with(
-            False, 1692943200, False, True
-        )
+@enforce_types
+def test_submit_trueval_mocked_price_down(
+    agent, slot, predictoor_contract, monkeypatch
+):
+    monkeypatch.setattr("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_down)
+
+    result = agent.get_and_submit_trueval(slot, predictoor_contract.return_value, 60)
+    assert result == {"tx": "0x123"}
+    predictoor_contract.return_value.submit_trueval.assert_called_once_with(
+        False, 1692943200, False, True
+    )
 
 
-def test_submit_trueval_mocked_price_up(agent, slot, predictoor_contract):
-    with patch("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_up):
-        result = agent.get_and_submit_trueval(
-            slot, predictoor_contract.return_value, 60
-        )
-        assert result == {"tx": "0x123"}
-        predictoor_contract.return_value.submit_trueval.assert_called_once_with(
-            True, 1692943200, False, True
-        )
+@enforce_types
+def test_submit_trueval_mocked_price_up(agent, slot, predictoor_contract, monkeypatch):
+    monkeypatch.setattr("ccxt.kraken.fetch_ohlcv", mock_fetch_ohlcv_up)
+
+    result = agent.get_and_submit_trueval(slot, predictoor_contract.return_value, 60)
+    assert result == {"tx": "0x123"}
+    predictoor_contract.return_value.submit_trueval.assert_called_once_with(
+        True, 1692943200, False, True
+    )
 
 
+@enforce_types
 def test_take_step(slot, agent):
     mocked_env = {
         "SLEEP_TIME": "1",
@@ -68,6 +77,7 @@ def test_take_step(slot, agent):
     ps_mock.assert_called_once_with(slot)
 
 
+@enforce_types
 def test_run(slot, agent):
     mocked_env = {
         "SLEEP_TIME": "1",

